### PR TITLE
Added abstract FileManager

### PR DESCRIPTION
**Pull request by dommi2212**
[+] FileManager.java
[*] Changed LocationManager to new system
[*] Slightly changed Main

*Edit*:
Added GameState
[+] GameState.java
[*] Change Main.java to support GameStates

### CODE REVIEW

**Comment by Rincewind34** [comment]
Please observe our conventions. But good job xD

**Comment by Janhektor** [comment]
Right, Rincewind34.
The code style guidelines are important - it's pays off later.

dommi2212, it would be great if you change your code to observe the conventions.

Regards,
Janhektor

**Comment by dommi2212** [comment]
If would be very happy, if you told me, which guiedline I didn't "respect" ;)

**Comment by Janhektor** [comment]
Okay, I'll comment some pieces of your code.

File: src/de/janhektor/community/config/FileManager.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		this.file = file;
+		this.config = YamlConfiguration.loadConfiguration(file);
+		init();
+	}
+	
+	/**
+	 * Creates a new file every time. Use {@link #createFileIfNotExistst()} to create the file only if it doesn't exist.
+	 */
+	protected void createFile() {
+		if(!file.getParentFile().exists()) file.getParentFile().mkdir();
```
Comment (Janhektor) [inline src/de/janhektor/community/config/FileManager.java]: Please type a space between if and the first opening bracket.
<b>And</b>: Use a block for your if-statement.
After observing the conventions this line may look like this:
if (!file.getParentFile().exists()) {
file.getParentFile().mkdir();
}

File: src/de/janhektor/community/config/FileManager.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Creates the file only if it doesn't exist.
+	 */
+	protected void createFileIfNotExists() {
+		if(!file.exists()) createFile();
```
Comment (Janhektor) [inline src/de/janhektor/community/config/FileManager.java]: See above.
Please use a space between if and the first opening bracket.
Use a block.

File: src/de/janhektor/community/config/FileManager.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	 * a: 1
+	 * b: 2
+	 * c: 3
+	 *
+	 * @param paths the paths to set.
+	 * @param values the values to set.
+	 * @throws IllegalArgumentException if the size of the lists isn't equal.
+	 */
+	public void set(List<String> paths, List<Object> values) {
+		if(paths.size() != values.size()) throw new IllegalArgumentException("Size of paths and values has to be equal!");
```
Comment (Janhektor) [inline src/de/janhektor/community/config/FileManager.java]: See above.

File: src/de/janhektor/community/config/FileManager.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		
+		for(int i = 0; i < paths.size(); i++) {
```
Comment (Janhektor) [inline src/de/janhektor/community/config/FileManager.java]: See above; use a space between for and the first opening bracket.

File: src/de/janhektor/community/config/LocationManager.java [hunk 5]
```diff
@@ -33,50 +31,11 @@ public Location getLocation(String key) {
...
-				this.file.createNewFile();
-			}
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
-		
-		if (this.configuration == null) {
-			this.configuration = YamlConfiguration.loadConfiguration(this.file);
-		}
+		set(key, loc.serialize());
```
Comment (Janhektor) [inline src/de/janhektor/community/config/LocationManager.java]: Use this.set instead of set (Google code style for Java).

**Comment by dommi2212** [comment]
Ty for information..

**Comment by TheRealJasmin** [comment]
How can you know GAmeStates when we arent sure how this game will work yet?

**Comment by Janhektor** [comment]
Sorry, but GameStates are a part of my second tutorial.
Please create a new pull request without your GameState-class.

**Comment by dommi2212** [comment]
@TheRealJasmin There always has to be sort of "Rebelism" to make progress... Apart from that Han pounted out this steps in the first video

@Janhektor I can't know that ^^ But I will create the new Request soon... (-.-)

**Comment by Janhektor** [comment]
OK, thank you.
Sorry for rejecting - It was accident that I already had a GameState manager.
Your work FileManager is good, it would be cool if you create a new pull request containing it.

**Pull request closed without merging**